packages/core/src/main/notification/NotificationService.ts

```diff
@@ -258,25 +257,30 @@ export class NotificationService extends EventEmitter {
         const groupIdStr = Encoder.toBase64(newGroupId).asString;
         // The groupId can then be sent back to the consumer
         return {
-          event,
+          event: {...event, data: groupIdStr},
           mappedEvent: ConversationMapper.mapConversationEvent({...event, data: groupIdStr}, source),
         };
 
       case Events.CONVERSATION_EVENT.MLS_MESSAGE_ADD:
         const encryptedData = Decoder.fromBase64(event.data).asBytes;
 
-        const groupId = await this.getUint8ArrayFromConversationGroupId(
+        const groupId = await this.getGroupIdFromConversationId(
           event.qualified_conversation ?? {id: event.conversation, domain: ''},
         );
+        const groupIdBytes = Decoder.fromBase64(groupId).asBytes;
 
         // Check if the message includes proposals
-        const {proposals, commitDelay, message} = await this.mlsService.decryptMessage(groupId, encryptedData);
-        if (proposals.length > 0) {
+        const {proposals, commitDelay, message} = await this.mlsService.decryptMessage(groupIdBytes, encryptedData);
+        if (typeof commitDelay === 'number' || proposals.length > 0) {
+          // we are dealing with a proposal, add a task to process this proposal later on
+          // Those proposals are stored inside of coreCrypto and will be handled after a timeout
           await this.handlePendingProposals({
-            groupId: groupId.toString(),
+            groupId,
             delayInMs: commitDelay ?? 0,
             eventTime: event.time,
           });
+          // This is not a text message, there is nothing more to do
```

Comment:
what do you mean by this comment?

Reply:
since it's a proposal message, there is no content to the actual event so we can just return the raw event. 
Would you suggest another wording?

Reply:
maybe to me its more clear if message is renamed to proposalMessage on line 273? and then this comment is removed. its fine either way

Reply:
well this `message` _can_ be a text message. But if there are `delay` or `proposal` content, then we know it's not a text message and we can early return avoiding the text-specific code to execute